File: website-main/scripts/generate-runtime-config.js
const fs = require('fs');
const path = require('path');

const rootDir = path.resolve(__dirname, '..');
const envPath = path.join(rootDir, '.env');
const outputPath = path.join(rootDir, 'js', 'runtime-config.js');

function parseEnv(filePath) {
  const file = fs.existsSync(filePath) ? fs.readFileSync(filePath, 'utf8') : '';
  const env = {};

  for (const line of file.split(/\r?\n/)) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith('#')) continue;

    const separatorIndex = trimmed.indexOf('=');
    if (separatorIndex === -1) continue;

    const key = trimmed.slice(0, separatorIndex).trim();
    const value = trimmed.slice(separatorIndex + 1).trim();
    env[key] = value.replace(/^['"]|['"]$/g, '');
  }

  return env;
}

const fileEnv = parseEnv(envPath);
const env = { ...fileEnv, ...process.env };

const runtime = {
  SUPABASE_URL: env.SUPABASE_URL || '',
  SUPABASE_ANON_KEY: env.SUPABASE_PUBLISHABLE_KEY || env.SUPABASE_ANON_KEY || ''
};

const missing = Object.entries(runtime)
  .filter(([, value]) => !value)
  .map(([key]) => key);

if (missing.length > 0) {
  console.warn(`Warning: missing values for ${missing.join(', ')}.`);
}

const content = `window.LIGHTSOUT_ENV = ${JSON.stringify(runtime, null, 2)};\n`;
fs.writeFileSync(outputPath, content, 'utf8');
console.log(`Generated runtime config at ${path.relative(rootDir, outputPath)}`);

File: website-main/scripts/generate-runtime-config.py
from pathlib import Path

root = Path(__file__).resolve().parent.parent
env_path = root / '.env'
output_path = root / 'js' / 'runtime-config.js'

config = {
    'SUPABASE_URL': '',
    'SUPABASE_ANON_KEY': '',
    'ADMIN_PASSWORD': '',
    'WRITER_PASSWORD': ''
}

if env_path.exists():
    for line in env_path.read_text(encoding='utf-8').splitlines():
        stripped = line.strip()
        if not stripped or stripped.startswith('#') or '=' not in stripped:
            continue
        key, value = stripped.split('=', 1)
        key = key.strip()
        value = value.strip().strip('"\'')
        if key in config:
            config[key] = value

output_path.write_text(
    'window.LIGHTSOUT_ENV = ' + str({k: v for k, v in config.items() if v}).replace("'", '"') + ';\n',
    encoding='utf-8'
)

print(f'Generated runtime config at {output_path.relative_to(root)}')
